File: php/testcode/benchmark_test_00358.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00358(BenchmarkRequest $req): BenchmarkResponse {
    $filename = $req->param('file');
    $allowed = ['readme.txt', 'license.txt', 'changelog.txt'];
    if (!in_array($filename, $allowed, true)) {
        return BenchmarkResponse::badRequest('not allowed');
    }
    $content = file_get_contents('/var/app/public/' . $filename);
    return BenchmarkResponse::ok($content);
}

File: rust/testcode/benchmark_test_00358.rs
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let filename = req.param("filename");
    let content = req.body_str();
    let ext = filename.rsplit('.').next().unwrap_or("");
    if ext == "txt" || ext == "csv" || ext == "log" {
        let path = format!("uploads/{}", filename);
        let _ = std::fs::write(&path, content.as_bytes());
        super::shared::BenchmarkResponse::ok(&format!("Saved: {}", path))
    } else {
        super::shared::BenchmarkResponse::bad_request("Unsupported file type")
    }
}
